src/components/WalletSelector/WalletSelector.tsx

```diff
@@ -74,28 +71,22 @@ function WalletSelector() {
   useEffect(() => {
     if (account && isPending && signer) {
       signMessageAndAuthenticate(account, signer)
-        .then((jwt) => {
-          logIn(jwt);
-          navigate('/welcome');
-        })
```

Comment:
we don't need to navigate here since redirect will be handled by the parent once context authenticates